fdmi/plugins/fdmi_sample_plugin.c

```diff
@@ -0,0 +1,426 @@
+	buffer[2 * len - 2] = '\0';
+	return buffer;
+}
+
+M0_INTERNAL void m0_dump_m0_fol_rec_to_json(struct m0_fol_rec *rec)
+{
+	struct m0_fol_frag *frag;
+	int i;
+
+	m0_tl_for(m0_rec_frag, &rec->fr_frags, frag) {
+		struct m0_fop_fol_frag *fp_frag = frag->rp_data;
+		struct m0_cas_op *cas_op = fp_frag->ffrp_fop;
+		struct m0_cas_recv cg_rec = cas_op->cg_rec;
+		struct m0_cas_rec *cr_rec = cg_rec.cr_rec;
+
+		for (i = 0; i < cg_rec.cr_nr; i++) {
+			int len = 0;
+
+			m0_console_printf("{ ");
+
+			len = sizeof(struct m0_fid);
+			m0_console_printf("\"fid\": \"%s\", ",
+					  to_hex((void *)&cas_op->cg_id.ci_fid, len));
+
+			len = cr_rec[i].cr_key.u.ab_buf.b_nob;
+			m0_console_printf("\"cr_key\": \"%s\", ",
+					  to_hex(cr_rec[i].cr_key.u.ab_buf.b_addr, len));
+
+			len = cr_rec[i].cr_val.u.ab_buf.b_nob;
+			if (len > 0) {
+				m0_console_printf("\"cr_val\": \"%s\"",
+						  to_hex(cr_rec[i].cr_val.u.ab_buf.b_addr, len));
+			} else {
+				m0_console_printf("\"cr_val\": \"0\"");
+			}
+			m0_console_printf(" }\n");
+
+		}
+	} m0_tl_endfor;
+}
+
+static void fsp_usage(void)
+{
+	fprintf(stderr,
+		"Usage: fdmi_sample_plugin "
+		"-l local_addr -h ha_addr -p profile_fid -f process_fid "
+		"-g fdmi_plugin_fid\n"
+		"Use -? or -i for more verbose help on common arguments.\n"
+		"Usage example for common arguments: \n"
+		"fdmi_sample_pluginm0sched -l 192.168.52.53@tcp:12345:4:1 "
+		"-h 192.168.52.53@tcp:12345:1:1 "
+		"-p 0x7000000000000001:0x37 -f 0x7200000000000001:0x19"
+		"-g 0x6c00000000000001:0x51"
+		"\n");
+}
+
+/**
+ * @retval 0      Success.
+ * @retval -Exxx  Error.
+ */
+static int fsp_args_parse(struct m0_fsp_params *params, int argc, char ** argv)
+{
+	int rc = 0;
+
+	params->spp_local_addr 	= NULL;
+	params->spp_hare_addr   = NULL;
+	params->spp_profile_fid = NULL;
+	params->spp_process_fid = NULL;
+	params->spp_fdmi_plugin_fid_s = NULL;
+
+	rc = M0_GETOPTS("fdmi_sample_plugin", argc, argv,
+			M0_HELPARG('?'),
+			M0_VOIDARG('i', "more verbose help",
+				   LAMBDA(void, (void) {
+						   fsp_usage();
+						   exit(0);
+					   })),
+			M0_STRINGARG('l', "Local endpoint address",
+				     LAMBDA(void, (const char *string) {
+						     params->spp_local_addr = (char*)string;
+					     })),
+			M0_STRINGARG('h', "HA address",
+				     LAMBDA(void, (const char *str) {
+						     params->spp_hare_addr = (char*)str;
+					     })),
+			M0_STRINGARG('f', "Process FID",
+				     LAMBDA(void, (const char *str) {
+						     params->spp_process_fid = (char*)str;
+					     })),
+			M0_STRINGARG('p', "Profile options for client",
+				     LAMBDA(void, (const char *str) {
+						     params->spp_profile_fid = (char*)str;
+					     })),
+			M0_STRINGARG('g', "FDMI plugin fid",
+				     LAMBDA(void, (const char *str) {
+						     params->spp_fdmi_plugin_fid_s =
+							     (char*)str;
+					     })));
+	if (rc != 0)
+		return M0_ERR(rc);
+
+        /* All mandatory params must be defined. */
+	if (rc == 0 &&
```

Comment:
Please see the codacy warning and fix it.

Reply:
Cannot find that warning of codacy.